hazelcast-client-new/src/main/java/com/hazelcast/client/cache/impl/AbstractClientInternalCacheProxy.java

```diff
@@ -80,7 +95,16 @@
     private static final long MAX_COMPLETION_LATCH_WAIT_TIME = TimeUnit.MINUTES.toMillis(5);
     private static final long COMPLETION_LATCH_WAIT_TIME_STEP = TimeUnit.SECONDS.toMillis(1);
 
-    protected final ClientNearCache<Data, Object> nearCache;
+    protected final ILogger logger = Logger.getLogger(getClass());
+
+    protected final HazelcastClientCacheManager cacheManager;
+    protected final NearCacheManager nearCacheManager;
+    // Object => Data or <V>
+    protected NearCache<Data, Object> nearCache;
```

Comment:
`nearCache` field can be `final`. I guess there's no need to null it when destroyed. Otherwise it should be marked as `volatile`.